Automattic/pocket-casts-ios
# Remove redundant "Any SDK" settings for code signing identity

They were redundant because, by being "Any SDK" and being the only specialization, they acted as a catch all.

On top of that, some settings had "iOS Distribution" as the default and "iOS Developer" in the "Any SDK" catch-all specialization, which was confusing.

## To test

Try to build on device or to archive.

<img width="566" alt="Screenshot 2023-08-10 at 5 10 58 pm" src="https://github.com/Automattic/pocket-casts-ios/assets/1218433/e719c731-dc89-45b7-a5de-51abcf665bd0">

If Xcode can't do it because of lack of access to the certs, apply this patch:

```
--- a/fastlane/Fastfile
+++ b/fastlane/Fastfile
@@ -356,6 +356,8 @@ platform :ios do
       }
     )
 
+    UI.user_error!('Forcing early return for testing')
+
     secrets_dir = File.join(Dir.home, '.configure', 'pocketcasts-ios', 'secrets')
     testflight(
       skip_waiting_for_build_processing: true,
```

And verify `bundle exec fastlane build_and_upload_app_store_connect skip_confirm:true skip_prechecks:true` succeeds.

## Checklist
